x-pack/plugins/lens/public/indexpattern_datasource/index.ts

```diff
@@ -48,8 +48,11 @@ export class IndexPatternDatasource {
       } = await import('../async_services');
       return core
         .getStartServices()
-        .then(([coreStart, { data, indexPatternFieldEditor, uiActions }]) => {
-          data.fieldFormats.register([getSuffixFormatter(data.fieldFormats.deserialize)]);
+        .then(([coreStart, { indexPatternFieldEditor, uiActions, data }]) => {
+          const suffixFormatter = getSuffixFormatter(data.fieldFormats.deserialize);
+          if (!dataSetup.fieldFormats.has(suffixFormatter.id)) {
+            dataSetup.fieldFormats.register([suffixFormatter]);
```

Comment:
Why you are registering this to the upper `data` plugin rather than the inner one? Can you explain with a comment?

Reply:
In this pr, I fixed the type definition of field formats registry so that you don't have `register()` function on the start contract anymore. It was available prior to this pr because of an issue in type definitions. The only consumer who used `register` on a start contract was the lens.

![Screen Shot 2021-07-28 at 13 12 04](https://user-images.githubusercontent.com/7784120/127313162-a3fa9303-dfa6-453e-9d18-6e17dfb8e56b.png)


I don't think this needs a code comment because there is no other option to call `register()` anymore. 

--- 

Longer-term we plan to make all registries allow new items to be registered **only** in a setup contract and only during the setup lifecycle.